### PR TITLE
Create tests for the first HTTP server

### PR DESCRIPTION
**Pull request by 0xEG**

### CODE REVIEW

File: app01.test.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+'use strict';
+
+const app = require('./app');
+
+test('Query reguest validation', async () => {
```
Comment (JekRock) [inline app01.test.js]: The `test` function takes a string as a first argument. That string should contain a short explanation of what a specific test does.
Could you change the description to match what test does (and the same for the second test)?

File: app01.test.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+  const response = await app.inject({
+    method: 'GET',
+    url: '/?name=Ted',
```
Comment (JekRock) [inline app01.test.js]: [Fastify's library](https://github.com/fastify/light-my-request#api) used for testing accepts a property called `query` that should be used to pass query parameters for a request.
Could you use this property to pass query parameters?

File: app02.test.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+test('Query reguest validation', async () => {
+  const response = await app.inject({
+    method: 'GET',
+    url: '/',
+  });
+
+  expect(response.statusCode).toBe(200);
+  expect(response.payload).toBe('Hello world');
+});
```
Comment (JekRock) [inline app02.test.js]: In this scenario, there is no need to split tests into multiple files. You can place as many tests as you need into a single file as long as it makes sense.
Also, file names should be meaningful. Something like `hello.test.js` or `hello-app.test.js` will work for these tests.

**Review by JekRock — CHANGES_REQUESTED** [review]

File: app01.test.js (hunk 2 above)
Comment (0xEG) [reply app01.test.js]: Fixed

File: app01.test.js (hunk 1 above)
Comment (0xEG) [reply app01.test.js]: Were done in both files

File: app02.test.js (hunk 3 above)
Comment (0xEG) [reply app02.test.js]: Done

Comment (JekRock) [reply app02.test.js]: Do you want to move the tests to a single file?

File: app01.test.js (hunk 2 above)
Comment (JekRock) [reply app01.test.js]: It doesn't look like it was changed

Comment (0xEG) [reply app01.test.js]: A property (query) from Fastify`s library was used to pass both tests:
     expect(response.query).toBe('Hello Ted');
     expect(response.query).toBe('Hello world');

File: app02.test.js (hunk 3 above)
Comment (0xEG) [reply app02.test.js]: If I should - I do, but for this case we could leave it in two files.

Comment (JekRock) [reply app02.test.js]: There is no need to separate them. They test the same module.
Usually, you have a file per module that is being tested

File: app01.test.js (hunk 2 above)
Comment (JekRock) [reply app01.test.js]: Please look at what my comment was pointing to in the code.
I meant the `app.inject` method, not `response.query`.

Currently, you're passing the query parameter as a string in the `url` property. Fastify accepts a property called `query` for passing query parameters

File: app02.test.js (hunk 3 above)
Comment (0xEG) [reply app02.test.js]: Done.

File: app01.test.js (hunk 2 above)
Comment (0xEG) [reply app01.test.js]: Done.

**Review by JekRock — APPROVED** [review]

**Pull request merged**